### PR TITLE
feat: add new package for griffel Devtools extension

### PR DESCRIPTION
**Pull request by YuanboXue-Amber**
Add a new package @griffel/devtools to hold UI for griffel devtools.

- package location: _packages/devtools_
- manifest and entry files are located in _packages/devtools/public_
- package is generated using nx template for web apps. Changes to template:
     - custom _webpack.config.js_ to bundle  _packages/devtools/public_ with code
     - added `pack-extension` command to zip extension

Built extension is located in _dist/packages/devtools_

Currently extension contains only a default message:
Screenshot 2022-04-19 at 12 40 39

Follow up plan:
- add storybook for extension components
- build real extension UI
- figure out how to publish extension
    - extension manifest.json must be bumped with package.json

### CODE REVIEW

File: packages/core/.babelrc [hunk 1]
```diff
@@ -1,5 +1,6 @@
 {
   "plugins": ["annotate-pure-calls"],
+  "presets": ["@babel/preset-typescript"],
```
Comment (YuanboXue-Amber) [inline packages/core/.babelrc:3]: Reason for this change (open for better solutions):
When I start to use `makeStyles` in devtools, I get the following error during build. And the error only goes away when I add babel TS preset in both @griffel/core and @griffel/react
Screenshot 2022-04-19 at 12 45 06

File: packages/devtools/.eslintrc.json [hunk 2]
```diff
@@ -0,0 +1,30 @@
+{
+  "env": {
+    "webextensions": true
+  },
+  "extends": ["plugin:@nrwl/nx/react", "../../.eslintrc.json"],
+  "ignorePatterns": ["!**/*"],
+  "rules": {
+    "import/no-extraneous-dependencies": ["error", { "devDependencies": true }]
```
Comment (YuanboXue-Amber) [inline packages/devtools/.eslintrc.json:8]: I added override of `no-extraneous-dependencies` rule. Without it, I get error:
image

File: packages/core/.babelrc (hunk 1 above)
Comment (layershifter) [reply packages/core/.babelrc:3]: We agreed on a clearer option:

```
{
  "plugins": ["annotate-pure-calls"],
  "env": {
    "development": {
      "presets": ["@nrwl/react/babel"]
    },
    "test": {
      "presets": ["@nrwl/react/babel"]
    }
  }
}
```

File: packages/devtools/.eslintrc.json (hunk 2 above)
Comment (layershifter) [reply packages/devtools/.eslintrc.json:8]:
```suggestion
```

We don't need need it, but we need it to specify `react` & `react-dom` to `dependencies` in `packages/devtools/package.json`:

```
    "react": "17.0.2",
    "react-dom": "17.0.2"
```

File: packages/devtools/.eslintrc.json [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    {
+      "files": ["*.js", "*.jsx"],
+      "rules": {}
+    },
+    {
+      "files": ["webpack.config.js"],
+      "rules": {
+        "import/no-extraneous-dependencies": "off"
+      }
+    }
```
Comment (layershifter) [inline packages/devtools/.eslintrc.json:28]:
```suggestion
    }
```

File: packages/devtools/README.md [hunk 4]
```diff
@@ -0,0 +1,9 @@
+# Griffel Devtools
+
+Chrome devtools panel extension for debugging griffel.
+
+### Contribution
+
+- run `yarn devtools:watch` from workspace root
```
Comment (layershifter) [inline packages/devtools/README.md:7]: Let's update this based on another suggestion

File: packages/devtools/package.json [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+  "version": "0.0.1",
+  "description": "Griffel chrome devtools extension",
+  "license": "MIT",
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/microsoft/griffel"
+  },
+  "sideEffects": false,
+  "dependencies": {
+    "@griffel/react": "^1.0.3"
```
Comment (layershifter) [inline packages/devtools/package.json:12]:
```suggestion
    "@griffel/react": "^1.0.3",
    "react": "17.0.2",
    "react-dom": "17.0.2"
```

File: packages/devtools/project.json [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+      "executor": "@nrwl/web:webpack",
+      "outputs": ["{options.outputPath}"],
+      "defaultConfiguration": "production",
+      "options": {
+        "compiler": "babel",
+        "outputPath": "dist/packages/devtools",
+        "index": "packages/devtools/src/index.html",
+        "main": "packages/devtools/src/index.tsx",
+        "tsConfig": "packages/devtools/tsconfig.lib.json",
+        "webpackConfig": "packages/devtools/webpack.config.js",
```
Comment (layershifter) [inline packages/devtools/project.json:17]:
```suggestion
```

File: packages/devtools/project.json [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+      }
+    },
+    "test": {
+      "executor": "@nrwl/jest:jest",
+      "outputs": ["coverage/packages/devtools"],
+      "options": {
+        "jestConfig": "packages/devtools/jest.config.js",
+        "passWithNoTests": true
+      }
+    }
```
Comment (layershifter) [inline packages/devtools/project.json:62]:
```suggestion
    },
    "type-check": {
      "executor": "@nrwl/workspace:run-commands",
      "options": {
        "cwd": "packages/devtools",
        "commands": [{ "command": "tsc -b --pretty" }],
        "outputPath": []
      }
    }
```

Comment (layershifter) [reply packages/devtools/project.json:62]: Let's add proper typechecking

File: packages/devtools/tsconfig.lib.json [hunk 8]
```diff
@@ -0,0 +1,18 @@
+{
+  "extends": "./tsconfig.json",
+  "compilerOptions": {
+    "outDir": "../../dist/out-tsc",
+    "types": ["node"]
```
Comment (layershifter) [inline packages/devtools/tsconfig.lib.json:5]:
```suggestion
    "types": ["chrome", "node", "environment"]
```

File: packages/devtools/tsconfig.spec.json [hunk 9]
```diff
@@ -0,0 +1,19 @@
+{
+  "extends": "./tsconfig.json",
+  "compilerOptions": {
+    "outDir": "../../dist/out-tsc",
+    "module": "commonjs",
+    "types": ["jest", "node"]
```
Comment (layershifter) [inline packages/devtools/tsconfig.spec.json:6]:
```suggestion
    "types": ["chrome", "jest", "node", "environment"]
```

File: packages/devtools/project.json [hunk 10]
```diff
@@ -0,0 +1,86 @@
...
+          {
+            "glob": "packages/devtools/public",
+            "input": ".",
+            "output": "."
+          }
+        ]
+      },
+      "configurations": {
+        "production": {
+          "optimization": true,
```
Comment (layershifter) [inline packages/devtools/project.json:28]:
```suggestion
          "buildLibsFromSource": false,
          "optimization": true,
```

**Review by layershifter — APPROVED** [review]

**Pull request merged**